packages/backend/test/peripherals/coingecko/CoingeckoQueryService.test.ts

```diff
@@ -0,0 +1,49 @@
+import { UnixTime } from "@l2beat/common";
+import { expect } from "earljs";
+
+import { CoingeckoQueryService, generateTimestampsList, Granularity } from "../../../src/peripherals/coingecko/CoingeckoQueryService";
+
+describe(CoingeckoQueryService.name, () => {
+})
+
+describe(generateTimestampsList.name, () => {
+    describe("hourly", () => {
+        const GRANULARITY = Granularity.Hourly
+        const FROM = new UnixTime(1643806800) //Wed Feb 02 2022 13:00:00 GMT+0000
+        const TO = new UnixTime(1643814000) //Wed Feb 02 2022 15:00:00 GMT+0000
+
+        const RESULT = [new Date(1643806800000), new Date(1643810400000), new Date(1643814000000)]
+
+        it("empty if FROM greater than TO", () => {
+            const result = generateTimestampsList(TO,FROM,GRANULARITY)
+            expect(result).toEqual([])
+        })
+
+        it("empty if FROM is not a full hour", () => {
```

Comment:
In this case shouldn't we start from the next full hour after `FROM`. Similarity for `TO` if it isn't a full hour, shouldn't we use the previous full hour before it? I think it makes more sense then requiring precision. If we require precision then throwing is definitely the answer, because an empty array is an unexpected, undetected, silent failure.

Reply:
I think I will make it both ways inclusive, 
e.g. from = 14:15, to = 15:20, granularity = 'hourly'
returns [14:00,15:00,16:00]

@sz-piotr let me know what u think